include/net/nrf_cloud.h

```diff
@@ -822,13 +822,30 @@ int nrf_cloud_modem_fota_completed(const bool fota_success);
 /**
  * @brief Retrieve the current device ID.
  *
- * @param[in,out] id_buf Buffer to receive the device ID.
- * @param[in] id_len     Size of buffer (NRF_CLOUD_CLIENT_ID_MAX_LEN).
+ * @param[in,out] id_buf Buffer to receive the device ID as a null-terminated string.
+ * @param[in] id_buf_sz  Size of buffer, maximum size is NRF_CLOUD_CLIENT_ID_MAX_LEN + 1.
+ *
+ * @retval 0         If successful.
+ * @retval -EMSGSIZE The provided buffer is too small.
+ * @retval -EIO      The client ID could not be initialized.
+ * @retval -ENXIO    The Kconfig option @kconfig{CONFIG_NRF_CLOUD_CLIENT_ID_SRC_RUNTIME} is enabled
+ *                   but the runtime client ID has not been set.
+ *                   See @ref nrf_cloud_client_id_runtime_set.
+ * @return A negative value indicates an error.
+ */
+int nrf_cloud_client_id_get(char *id_buf, size_t id_buf_sz);
+
+/**
+ * @brief Set the device ID at runtime.
+ *        Requires @kconfig{CONFIG_NRF_CLOUD_CLIENT_ID_SRC_RUNTIME} to be enabled.
+ *
```

Comment:
We should provide some guidance either here or in the library docs that explains it is up to the caller to manage the network connection when changing the device id. The device will fail to communicate with nrfcloud using this device id if it is not already onboarded / registered using this device id.

Reply:
> The device will fail to communicate with nrfcloud using this device id if it is not already onboarded / registered using this device id.

this is true for any `CONFIG_NRF_CLOUD_CLIENT_ID_SRC_` that is selected.
you already have a ticket for this doc improvement 😄 
IRIS-5868

> it is up to the caller to manage the network connection when changing the device id.

added a note